pkgs/by-name/po/poetry/package.nix

```diff
@@ -11,6 +11,17 @@ let
     {
       poetry = self.callPackage ./unwrapped.nix { };
 
+      # Poetry 2.1.4 officially requires virtualenv >=20.26.6, <20.33.0
+      # otherwise will be incompatible with python312
```

Comment:
It would be nice to link to https://github.com/python-poetry/poetry/issues/10490